README.md

```diff
@@ -15,17 +15,28 @@ Implement the tool `st` that does exact pattern matching using a suffix tree. Te
 
 ## Specify if you have used a linear time or quadratic time algorithm.
 
+I have implemented McCreight's O(n) algorithm.
+
 ## Insights you may have had while implementing and comparing the algorithms.
 
+There is a lot to keep track of with the various nodes and suffix links. 
+
 ## Problems encountered if any.
 
+I used a lot of time to understand the algorithm. 
+After, it was difficult to implement in python because some things in C are so different.
+
 ## Correctness
 
-*Describe experiments that verifies the correctness of your implementations.*
+I used all the fasta and fastq files from the previous project and tested wheter the output was as supposed.
 
 ## Running time
 
-*Describe experiments that verifies that your implementation of `st` uses no more time than O(n) or O(n²) (depending on the algorithm) for constructing the suffix tree and no more than O(m) for searching for a given read in it. Remember to explain your choice of test data. What are “best” and “worst” case inputs?*
+I simulated dna sequences with lengths ranging from 0 to 1000 bases. Then I recorded the time it took to build a suffix tree for each (see time.py). The sequence length was then plotted against the time recorded (see figure below).
+
+![](figs/time.png)
+
+The running time seems linear, however, there is some noise that seems random. To solve it, I put a delay between each sequence and it didn't solve the problem. I tested 1000 times with the same sequence and the same pattern was seen again (see the figure below). Therefore, it must be something with my computer. I don't know.
```

Comment:
I don't quite see what putting delays in the code would do, but if you are worried about stochastic noise, then using larger samples, or averaging over more runs, is the way to go. Measurements can be very noisy, but averages tend to be fine (central limit theorem and all that jive).

Stochastic variation doesn't have to be independent on data, however. Doing 1000 runs with the same sequence will average out variation caused by whatever else is happening on your computer (caching warmup, other processes, whatever it might be), but it won't do anything to handle that some data hits best cases and other worst cases. To average over that, you want to sample 1000 different sequences instead.

But it looks ok.